lib/ansible/modules/commands/psexec.py

```diff
@@ -0,0 +1,527 @@
+        module.fail_json(msg='Failed to authenticate over SMB: %s'
+                             % to_native(exc))
+    except SMBResponseException as exc:
+        module.fail_json(msg='Received unexpected SMB response when opening '
+                             'the connection: %s' % to_native(exc))
+    except SMBException as exc:
+        module.fail_json(msg=to_native(exc), exception=exc)
+    except PDUException as exc:
+        module.fail_json(msg='Received an exception with RPC PDU message: %s'
+                             % to_native(exc))
+    except SCMRException as exc:
+        module.fail_json(msg='Received an exception when dealing with SCMR on '
+                             'the Windows host: %s' % to_native(exc))
+    except PypsexecException as exc:
+        module.fail_json(msg=to_native(exc), exception=exc)
+
+    # create PAExec service and run the process
+    result['changed'] = True
+    b_stdin = to_bytes(stdin, encoding='utf-8') if stdin else None
+    run_args = dict(
+        executable=executable, arguments=arguments, asynchronous=asynchronous,
+        load_profile=load_profile, interactive_session=interactive_session,
+        run_elevated=elevated, run_limited=limited,
+        username=process_username, password=process_password,
+        use_system_account=use_system, working_dir=working_directory,
+        priority=priority, show_ui_on_win_logon=show_ui_on_win_logon,
+        timeout_seconds=process_timeout, stdin=b_stdin
+    )
+    if not module.check_mode:
+        try:
+            win_client.create_service()
+        except PDUException as exc:
+            module.fail_json(msg='Failed to create PAExec service with PDU '
+                                 'fault: %s' % to_text(exc))
+        except SCMRException as exc:
+            module.fail_json(msg='Failed to create PAExec service with SCMR '
+                                 'Error: %s' % to_text(exc))
+        except PypsexecException as exc:
+            module.fail_json(msg=to_text(exc), exception=exc)
+        except SMBException as exc:
+            module.fail_json(msg=to_text(exc), exception=exc)
+
+        try:
+            proc_result = win_client.run_executable(**run_args)
+        except PAExecException as exc:
+            module.fail_json(msg='Received error when running remote process: '
+                                 '%s' % to_text(exc))
+        except PypsexecException as exc:
+            module.fail_json(msg=to_text(exc), exception=exc)
+        finally:
+            remove_artifacts(module, win_client)
+
+        if asynchronous:
+            result['pid'] = proc_result[2]
+        elif interactive:
+            result['rc'] = proc_result[2]
+        else:
+            result['stdout'] = proc_result[0]
+            result['stderr'] = proc_result[1]
+            result['rc'] = proc_result[2]
+
+    # close the SMB connection
+    try:
+        win_client.disconnect()
+    except SMBResponseException as exc:
```

Comment:
Same here with exception handling collapse. You could use something like `type(exc).__name__` in the string if you don't want the whole stacktrace but still want to know what it was- I'm guessing that's not useful info to an end user, but might be for troubleshooting.